### PR TITLE
RHDEVDOCS-5615: Updating the installing build strategies section

### PR DESCRIPTION
**Pull request by shipsing**
**Purpose**: To resolve this issue: https://issues.redhat.com/browse/RHDEVDOCS-5615

**Aligned team**: DevTools

**Version for cherrypicking**: build-docs-1.0

**Content for preview**:  
Installing sample build startegies: https://64609--docspreview.netlify.app/openshift-builds/latest/installing/installing-sample-build-strategies
Configuring builds: https://64609--docspreview.netlify.app/openshift-builds/latest/configuring/configuring-openshift-builds
Configuring build stratagies: https://64609--docspreview.netlify.app/openshift-builds/latest/configuring/configuring-build-strategies
Configuring build runs: https://64609--docspreview.netlify.app/openshift-builds/latest/configuring/configuring-build-runs
Build controller observability: https://64609--docspreview.netlify.app/openshift-builds/latest/observability/build-controller-observability
Overview of builds: https://64609--docspreview.netlify.app/openshift-builds/latest/about/overview-openshift-builds

**SME review**:  @apoorvajagtap @jkhelil @coreydaley @divyansh42 

**QE review**:  @jitendar-singh 

**Peer review**: @kelbrown20

### CODE REVIEW

**Comment by apoorvajagtap** [comment]
@shipsing I think we can replace the content of [source-to-image
](https://64609--docspreview.netlify.app/openshift-builds/latest/builds/installing-sample-build-strategies#about-s2i_installing-sample-build-strategies) as following:

~~~
This BuildStrategy is composed by [source-to-image](https://github.com/openshift/source-to-image) and [buildah](https://github.com/containers/buildah) in order to generate a Containerfile and prepare the application to be built later on with a builder.

s2i requires a specially crafted image, which can be passed as a parameter using the `spec.paramValues` field of Build CR.
~~~

@jkhelil I believe, we'll be just supporting [this sample](https://github.com/shipwright-io/build/blob/main/samples/buildstrategy/source-to-image/buildstrategy_source-to-image-redhat_cr.yaml)of s2i, and here I see buildah being used in combination. Please correct me in case the above doesn't align with our current support scope.

File: modules/ob-defining-the-build-specification.adoc [hunk 1]
```diff
@@ -6,7 +6,7 @@
 [id="ob-defining-the-build-specification_{context}"]
 = Build specification definition
 
-You can embed a complete `spec.build.spec` field into your `BuildRun` resource to build the image. The following example shows a `BuildRun` CR that configures the `spec.build.spec` field:
+You can embed a complete build specification into your `BuildRun` resource by using the `spec.build.spec` field. This specification embedding allows you to build an image without creating and maintaining a dedicated `Build` resource. The following example shows a `BuildRun` CR that configures the `spec.build.spec` field:
```
Comment (kelbrown20) [inline modules/ob-defining-the-build-specification.adoc:9]:
```suggestion
You can embed a complete build specification into your `BuildRun` resource using the `spec.build.spec` field. This specification embedding allows you to build an image without creating and maintaining a dedicated `Build` custom resource. The following example shows a `BuildRun` CR that configures the `spec.build.spec` field:
```

File: modules/ob-installing-buildah-strategy.adoc [hunk 2]
```diff
@@ -6,12 +6,7 @@
 [id="installing-buildah-strategy_{context}"]
 = Installing buildah strategy
 
-You can install the `buildah` cluster build strategy, which does not require a daemon to run and is available for use by unprivileged users.
-
-The `buildah` strategy is available for installation by using the following resources:
-
-* `buildah-shipwright-managed-push`: This cluster build strategy allows storing the build image locally to perform additional processing tasks, such as the image vulnerability scanning, before the push. It requires that the image is stored to an output directory that can be processed. The `$(params.shp-output-directory)` system parameter defines the output directory for processing. 
-* `buildah-strategy-managed-push`: This cluster build strategy contains the step to push the image to the target repository without any further processing.
+You can install the `buildah` cluster build strategy, which does not require a daemon to run and is available for use by unprivileged users. By using the `buildah-strategy-managed-push` resource for installing `buildah`, you can push the image to the target repository without any further processing.
```
Comment (kelbrown20) [inline modules/ob-installing-buildah-strategy.adoc:9]:
```suggestion
You can install the `buildah` cluster build strategy, which does not require a daemon to run and is available for unprivileged users. When using the `buildah-strategy-managed-push` resource for installing `buildah`, you can push the image to the target repository without any further processing.
```

**Review by kelbrown20 — COMMENTED** [review]
Hello! Great work on this, just had some nits and suggestions to consider!

File: builds/installing-openshift-builds.adoc [hunk 3]
```diff
@@ -17,6 +17,10 @@ As a cluster administrator, you can install {builds-title} on an {ocp-product-ti
 * You are logged in to the {ocp-product-title} cluster as an administrator.
 * Your cluster has the link:https://docs.openshift.com/container-platform/latest/installing/cluster-capabilities.html#marketplace-operator_cluster-capabilities[Marketplace capability] enabled or the Red Hat Operator catalog source configured manually.
 
+[NOTE]
+====
+As a dependency component, the {pipelines-operator} is automatically installed along with the {builds-operator} during installation. If you have already installed the {pipelines-operator}, {builds-shortname} uses that Operator for its functioning.
```
[the remark this answers is not in the document]
Comment (shipsing) [reply builds/installing-openshift-builds.adoc:22]: @kelbrown20, The provided suggestion does not convey the intended meaning. The builds Operator is not a dependency component. We want to highlight: The pipelines operator as a dependency component and convey that it is also installed along with the builds operator.

File: modules/ob-about-build-strategies.adoc [hunk 4]
```diff
@@ -8,19 +8,19 @@
...
-  name: buildpacks-v3
+  name: buildah
 spec:
   steps:
     - name: build-and-push
       image: docker.io/paketobuildpacks/builder-full:latest
-      env: 
+      env:
         - name: CNB_PLATFORM_API
           value: $(params.platform-api-version)
```
Comment (shipsing) [inline modules/ob-about-build-strategies.adoc:25]: @apoorvajagtap, Need your input to update this example with buildah.

Comment (apoorvajagtap) [reply modules/ob-about-build-strategies.adoc:25]: @shipsing the complete yaml needs to be replaced with the following:
~~~
apiVersion: shipwright.io/v1beta1
kind: ClusterBuildStrategy
metadata:
  name: buildah
spec:
  steps:
    - name: build-and-push
      image: quay.io/containers/buildah:v1.31.0
      workingDir: $(params.shp-source-root)
      command:
        - /bin/bash
       # ...
# ...
~~~

**Review by apoorvajagtap — CHANGES_REQUESTED** [review]

**Comment by jeana-redhat** [comment]
@shipsing Really nice update :smiley:  I have merged, but unfortunately it looks like you will need to do the cherrypick manually :weary:

**Pull request merged**